sdk/python/feast/infra/offline_stores/file.py

```diff
@@ -66,6 +74,35 @@ def _to_arrow_internal(self):
         df = self.evaluation_function()
         return pyarrow.Table.from_pandas(df)
 
+    def persist(self, storage: SavedDatasetStorage) -> "FileRetrievalJob":
+        assert isinstance(storage, SavedDatasetFileStorage)
+
+        filesystem, path = FileSource.create_filesystem_and_path(
+            storage.file_options.file_url, storage.file_options.s3_endpoint_override,
+        )
+
+        if path.endswith(".parquet"):
+            pyarrow.parquet.write_table(
+                self._to_arrow_internal(), where=path, filesystem=filesystem
+            )
+        else:
```

Comment:
probably makes sense to check it's not some other file format like .csv which we don't support

Reply:
In this `else` block we treat the destination as a directory, so it could be named even as `dir.csv` -> `dir.csv/` folder would be created.

Reply:
Added comment line